authentication/native/client.go

```diff
@@ -110,23 +110,28 @@ func (nac *nativeAuthClient) createNewToken() error {
 		return err
 	}
 
-	token := fmt.Sprintf("%s.%s.%d.%s", nac.encodedHost, lastHyperblock.Hash, nac.tokenExpiryInSeconds, nac.encodedExtraInfo)
+	token := &AuthToken{
+		ttl:       nac.tokenExpiryInSeconds,
+		host:      nac.host,
+		extraInfo: nac.extraInfo,
+		blockHash: lastHyperblock.Hash,
+		address:   nac.address,
+	}
 
-	signature, err := nac.txSigner.SignMessage([]byte(token), nac.skBytes)
+	token.signature, err = nac.signer.Sign(nac.privateKey, token.GetBody())
```

Comment:
Why did we switch to `Sign` from `SignMessage`?

Reply:
I wanted to have the same signer in both server & client.

Reply:
we should use `SignMessage` though everywhere, in a client setting **users should use known wallets (walletconnect, extension, web-wallet, etc)** - they should not provide private keys to any third party apps. Any secure wallet will only have endpoints to `signTransaction` or `signMessage` because having a sign endpoint for arbitrary data is very insecure.

The verification for the server in JS is also using signable messages: https://github.com/ElrondNetwork/native-auth-server/blob/main/src/native.auth.server.ts#L70 - so there's the compatibility part also

Reply:
changed back to SignMessage